dozer/cogs/moderation.py

```diff
@@ -292,88 +298,133 @@ async def on_message(self, message):
+        if message.content:
+            embed.add_field(name="Message Content:", value=str(message.content[0:1023]), inline=False)
+            if len(message.content) > 1024:
+                embed.add_field(name="Message Content Continued:", value=message.content[1024:2000], inline=False)
+        else:
+            embed.add_field(name="Message Content:", value="N/A", inline=False)
+        embed.set_footer(text="UserID:" + str(message.author.id))
         if message.attachments:
-            e.add_field(name="Attachments", value=", ".join([i.url for i in message.attachments]))
-        messagelogchannel = await self.edit_delete_config.query_one(guild_id=message.guild.id)
-        if messagelogchannel is not None:
-            channel = message.guild.get_channel(messagelogchannel.messagelog_channel)
+            embed.add_field(name="Attachments", value=", ".join([i.url for i in message.attachments]))
+        message_log_channel = await self.edit_delete_config.query_one(guild_id=message.guild.id)
+        if message_log_channel is not None:
+            channel = message.guild.get_channel(message_log_channel.messagelog_channel)
             if channel is not None:
-                await channel.send(embed=e)
+                await channel.send(embed=embed)
+
+    @Cog.listener()
+    async def on_raw_message_edit(self, payload):
+        """Logs message edits that are not currently in the bots message cache"""
+        if payload.cached_message:
+            return
+        Mchannel = self.bot.get_channel(int(payload.channel_id))
+        guild = Mchannel.guild
+        print(payload)
+        try:
+            content = payload.data['content']
+        except KeyError:
+            content = None
+        try:
+            author = payload.data['author']
+        except KeyError:
+            return
+        guild_id = str(guild.id)
+        channel_id = str(payload.channel_id)
+        user_id = str(author['id'])
+        if (self.bot.get_user(int(author['id']))).bot:
+            return  # Breakout if the user is a bot
+        message_id = str(payload.message_id)
+        link = f"https://discordapp.com/channels/{str(guild_id)}/{str(channel_id)}/{str(message_id)}"
+        mention = f"<@!{user_id}>"
+        avatar_link = f"http://cdn.discordapp.com/avatars/{str(user_id)}/{author['avatar']}.webp?size=1024"
+        embed = discord.Embed(title="Message Edited",
+                              description=f"[MESSAGE]({link}) From {mention}\nEdited In: {str(Mchannel.mention)}", color=0xFFC400)
+        embed.set_thumbnail(url=avatar_link)
+        embed.add_field(name="Original", value="N/A", inline=False)
+        if content:
+            embed.add_field(name="Edited", value=str(content[0:1023]), inline=False)
+            if len(content) > 1024:
+                embed.add_field(name="Edited Continued", value=str(content[1024:2000]), inline=False)
```

Comment:
Are the `str(...)` calls here necessary? F-strings format expressions with `str(...)` internally already.